src/append_only_data.rs

```diff
@@ -17,29 +17,44 @@ use std::{
     ops::Range,
 };
 
+/// Published sequenced AppendOnlyData.
```

Comment:
I think it's also worth adding a small module-level documentation block briefly explaining the differences between the `published` & `unpublished` and `sequenced` & `unsequenced`.